android/src/main/java/com/heanoria/library/reactnative/locationenabler/RNAndroidLocationEnablerModule.java

```diff
@@ -83,6 +83,14 @@ public void onActivityResult(Activity activity, int requestCode, int resultCode,
             if (resultCode == RESULT_OK ) {
                 promise.resolve("enabled");
             } else {
+                // Case for Android 10 bug: user pressed 'ok' but we receive RESULT_CANCELLED
+                // https://issuetracker.google.com/issues/140447198
+                if (getCurrentActivity() != null) {
+                    LocationManager locationManager = (LocationManager) getCurrentActivity().getSystemService(Context.LOCATION_SERVICE);
+                    if (locationManager != null && locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
```

Comment:
Please put this in a function like
```java
private Boolean isProviderEnabled() {
...
}
```
And then
```java
if (isProviderEnabled()) {
  promise.resolve("enabled");
} else {
  promise.reject(ERR_USER_DENIED_CODE, new RNAndroidLocationEnablerException("denied"));
}
```

Reply:
Agree, thanks for pointing it out.